core/unix/rseq_linux.c

```diff
@@ -269,6 +269,7 @@ rseq_shared_fragment_flushtime_update(dcontext_t *dcontext)
     rseq_clear_tls_ptr(dcontext);
 }
 
+#ifdef HAVE_RSEQ
```

Comment:
Actually, it looks like HAVE_RSEQ is only used to determine whether the rseq.h header is around, which then only determines whether the regression test is built: it does not affect whether the core handles an application using rseq.  The core rseq mangling code and all the code in rseq_linux.c is always enabled.  I believe the idea is that we want DR built on one machine to work on as many others as possible, rather than requiring a custom build for every variant of kernel feature or whatnot.

What is the top-level problem being solved here?  If it requires disabling or enabling the core handling rseq, probably a separate option should be put in place and maybe HAVE_RSEQ renamed to make it clearer what its consequences are.

Reply:
The underlying motivation here is to make it possible to record DR with the rr debugger. rr doesn't support rseq, but it also doesn't just return ENOSYS, it stops the program being recorded and outputs an error about the unsupported syscall.

Reply:
So maybe as suggested a separate control: a CMake option ENABLE_RSEQ_SUPPORT or sthg, and hopefully it can have as few ifdefs in the code as possible; combined with possibly renaming HAVE_RSEQ to HAVE_RSEQ_HEADER?